src/main/java/com/eventsphere/user/service/UserService.java

```diff
@@ -0,0 +1,225 @@
+package com.eventsphere.user.service;
+
+import com.eventsphere.user.exception.PasswordException;
+import com.eventsphere.user.exception.UserAlreadyExistsException;
+import com.eventsphere.user.exception.UserNotFoundException;
+import com.eventsphere.user.exception.UserNotValidException;
+import com.eventsphere.user.model.User;
+import com.eventsphere.user.model.dto.ChangePasswordDto;
+import com.eventsphere.user.model.dto.UserDto;
+import com.eventsphere.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Service class for managing user-related operations.
+ */
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    /**
+     * Retrieves a list of all users.
+     *
+     * @return List of {@link User} objects.
+     */
+    public List<User> getAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * Retrieves a specific user by their ID.
+     *
+     * @param id The ID of the user to retrieve.
+     * @return The {@link User} object.
+     * @throws UserNotFoundException if the user with the given ID is not found.
+     */
+    public User get(Long id) {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    }
+
+    /**
+     * Saves a user.
+     *
+     * @param user The {@link User} object to save.
+     * @return The saved {@link User} object.
+     * @throws UserNotValidException if the user data is invalid.
+     */
+    public User save(User user) {
+        User savedUser;
+
+        try {
+            user.setCreatedAt(null);
+            user.setUpdatedAt(null);
+
+            savedUser = userRepository.save(user);
+        } catch (RuntimeException ex) {
+            ex.printStackTrace();
+            throw new UserNotValidException("Invalid User data: " + ex.getMessage());
+        }
+
+        return savedUser;
+    }
+
+    /**
+     * Creates a new user.
+     *
+     * @param user The {@link User} object to create.
+     * @return The created {@link User} object.
+     * @throws UserAlreadyExistsException if a user with the same username or email already exists.
+     */
+    public User create(User user) {
+        User createdUser;
+
+        if (userRepository.existsByUsername(user.getUsername())) {
+            throw new UserAlreadyExistsException("This username is already registered");
+        } else if (userRepository.existsByEmail(user.getEmail())) {
+            throw new UserAlreadyExistsException("This email is already registered");
+        } else {
+            createdUser = save(user);
+        }
+
+        return createdUser;
+    }
```

Comment:
You have across this class useless declaration of user you can do instead something like this : 
``` java 
 public User create(User user) {


        if (userRepository.existsByUsername(user.getUsername())) {
            throw new UserAlreadyExistsException("This username is already registered");
        } else if (userRepository.existsByEmail(user.getEmail())) {
            throw new UserAlreadyExistsException("This email is already registered");
        } 
            
           return save(user);
    }
```